src/ucs/sys/topo.c

```diff
@@ -19,9 +53,125 @@ ucs_sys_bus_id_t ucs_sys_bus_id_unknown = { .domain   = 0xffff,
                                             .function = 0xff
                                           };
 
+static ucs_bus_id_bit_rep_t ucs_topo_get_bus_id_bit_repr(const ucs_sys_bus_id_t *bus_id)
+{
+    ucs_bus_id_bit_rep_t bit_rep = 0;
+
+    bit_rep |= bus_id->domain;
+    bit_rep <<= 24;
+    bit_rep |= bus_id->bus;
+    bit_rep <<= 16;
+    bit_rep |= bus_id->slot;
+    bit_rep <<= 8;
+    bit_rep |= bus_id->function;
+
+    return bit_rep;
```

Comment:
can this be written shorter by single expression?
or better yet, put ucs_sys_bus_id_t  in a union with uint64_t

Reply:
Probably missing something in the suggestion but the problem [here](https://github.com/openucx/ucx/pull/5194#issuecomment-633275564) is that `domain` member is needing promotion to `long int` type after left shifting by 24 bits. Is this problem solved by changing ucs_sys_bus_id_t from
```C
typedef struct ucs_sys_bus_id {
    uint16_t domain;   /* range: 0 to ffff */
    uint8_t  bus;      /* range: 0 to ff */
    uint8_t  slot;     /* range: 0 to 1f */
    uint8_t  function; /* range: 0 to 7 */
 } ucs_sys_bus_id_t;
```
to 
```C
typedef struct ucs_sys_bus_id {
    union {
        struct {
            uint16_t domain;   /* range: 0 to ffff */
            uint8_t  bus;      /* range: 0 to ff */
            uint8_t  slot;     /* range: 0 to 1f */
            uint8_t  function; /* range: 0 to 7 */
        };
        uint64_t dummy;
    };
} ucs_sys_bus_id_t;
```

Reply:
```
return ((uint64_t)bus_id->domain << 24) | (bus << 16) | 
            (slot << 8) | function;
```

the current code doesn't shift bit_rep right (it shifts too much)